src/opm/output/eclipse/LoadRestart.cpp

```diff
@@ -249,6 +250,48 @@ namespace {
         return { begin, end };
     }
 }
+// ---------------------------------------------------------------------
+class UDQVectors
+{
+public:
+    template <typename T>
+    using Window = boost::iterator_range<typename std::vector<T>::const_iterator>;
+
+    UDQVectors(std::shared_ptr<RestartFileView> rst_view) :
+        rstView_(rst_view)
+    {
+        const auto& intehead = rst_view->getKeyword<int>("INTEHEAD");
+        this->num_wells = intehead[VI::intehead::NWMAXZ];
+        this->num_groups = intehead[VI::intehead::NGMAXZ];
+    }
+
+    Window<double> next_dudw() {
+        return getDataWindow(this->rstView_->getKeyword<double>("DUDW"),
+                             this->num_wells, this->udq_well_index++);
+    }
+
+    Window<double> next_dudg() {
+        return getDataWindow(this->rstView_->getKeyword<double>("DUDG"),
+                             this->num_wells, this->udq_group_index++);
```

Comment:
Is `next_dudG` supposed to use `num_wells` as its window size?

Reply:
No .... :-)